runtimebp/cpu.go

```diff
@@ -12,88 +12,40 @@ import (
 // NumCPU returns the number of CPUs assigned to this running container.
 //
 // This is the container aware version of runtime.NumCPU.
-// It reads from the cgroup sysfs values.
+// It reads from the cgroup cpu.cfs_quota_us and cpu.cfs_period_us values
+// to determine the hard CPU limit of the container.
 //
 // If the current process is not running inside a container,
-// or for whatever reason we failed to read the cgroup sysfs values,
+// there is no cpu.cfs_quota_us, or if it failes to read the cgroup values,
 // it will fallback to runtime.NumCPU() instead.
-//
-// When fallback happens, it also prints the reason to stderr.
-func NumCPU() (n float64) {
+func NumCPU() float64 {
 	const (
 		quotaPath  = "/sys/fs/cgroup/cpu/cpu.cfs_quota_us"
 		periodPath = "/sys/fs/cgroup/cpu/cpu.cfs_period_us"
 	)
 
-	var err error
-	defer func() {
-		if err != nil || n <= 0 {
-			// Fallback and log to stderr.
-			fmt.Fprintf(
-				os.Stderr,
-				"NumCPU: falling back to use shares: %v, %v\n",
-				n,
-				err,
-			)
```

Comment:
I still think the logging about why the fallback happened is useful information and should not be removed. I'd suggest just remove the `numCPUSharesFallback` fallback and just do `n = float64(runtime.NumCPU())` on the next line instead.

Reply:
https://github.com/reddit/baseplate.go/pull/530 did the same thing so I don't think we need this PR any more?

Reply:
It's a good refactor, but this finishes the change by simplifying the code to be more explicit about the behavior.